objdiff-core/src/arch/ppc.rs

```diff
@@ -221,12 +221,21 @@ impl Arch for ArchPpc {
         }
     }
 
-    fn guess_data_type(&self, resolved: ResolvedInstructionRef) -> Option<DataType> {
+    fn guess_data_type(&self, resolved: ResolvedInstructionRef, bytes: &[u8]) -> Option<DataType> {
         if resolved.relocation.is_some_and(|r| r.symbol.name.starts_with("@stringBase")) {
+            // Pooled string.
             return Some(DataType::String);
         }
         let opcode = ppc750cl::Opcode::from(resolved.ins_ref.opcode as u8);
-        guess_data_type_from_load_store_inst_op(opcode)
+        if let Some(ty) = guess_data_type_from_load_store_inst_op(opcode) {
+            // Numeric type.
+            return Some(ty);
+        }
+        if bytes.len() >= 2 && bytes.iter().position(|&c| c == b'\0') == Some(bytes.len() - 1) {
```

Comment:
```suggestion
        if bytes.len() >= 2 && bytes.iter().position(|&c| c == b'\0') == bytes.last() {
```

I think this should work

Reply:
last would return the last byte in the slice, I think? This is comparing to the last index (len-1), not the value, since all the values were already read inside the loop.

The compiler also doesn't like last for some reason:
```
can't compare `std::option::Option<usize>` with `std::option::Option<&u8>`
the trait `PartialEq<std::option::Option<&u8>>` is not implemented for `std::option::Option<usize>`
but trait `PartialEq<deranged::OptionRangedUsize<_, _>>` is implemented for it
for that trait implementation, expected `deranged::OptionRangedUsize<_, _>`, found `std::option::Option<&u8>`
```

Reply:
I think my brain was melted by C++ iterators, nevermind